receiver/awscontainerinsightreceiver/internal/stores/podstore.go

```diff
@@ -472,13 +472,19 @@ func (p *PodStore) addStatus(metric CIMetric, pod *corev1.Pod) {
 					switch {
 					case containerStatus.State.Running != nil:
 						metric.AddTag(ci.ContainerStatus, "Running")
+						metric.AddField(ci.ContainerStatusRunning, 1)
```

Comment:
Another thing that I just thought of - We probably want to always emit metrics for all the states, always, where one of them will be a 1 and others will be a 0.

Consider a scenario where a container was running, crashed and then restarted to go back to a running state. So instead of the `running` metric being continuous with staying at 1, dropping to 0 and going back to 1, with the current implementation, we will see the metric with 1, then a missing chunk and then metric reappearing with value 1.

Moreover, for some of these states, if a container never transitions into those states, customers will never see those metrics, so writing their alarms may be tricky.

Reply:
Yeah I originally thought about that but forgot about it with all of the troubleshooting with getting any metrics to emit. Good callout. I will rework it to emit a zero where appropriate